cloud/services/object_storage_buckets_test.go

```diff
@@ -0,0 +1,275 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/linode/linodego"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	infrav1alpha1 "github.com/linode/cluster-api-provider-linode/api/v1alpha1"
+	"github.com/linode/cluster-api-provider-linode/cloud/scope"
+	"github.com/linode/cluster-api-provider-linode/mock"
+)
+
+func TestEnsureObjectStorageBucket(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name          string
+		bScope        *scope.ObjectStorageBucketScope
+		want          *linodego.ObjectStorageBucket
+		expectedError error
+		expects       func(mock *mock.MockLinodeObjectStorageClient)
+	}{
+		{
+			name: "Success - Successfully get the OBJ bucket",
+			bScope: &scope.ObjectStorageBucketScope{
+				Bucket: &infrav1alpha1.LinodeObjectStorageBucket{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-bucket",
+					},
+					Spec: infrav1alpha1.LinodeObjectStorageBucketSpec{
+						Cluster: "test-cluster",
+					},
+				},
+			},
+			want: &linodego.ObjectStorageBucket{
+				Label: "test-bucket",
+			},
+			expects: func(c *mock.MockLinodeObjectStorageClient) {
+				c.EXPECT().GetObjectStorageBucket(gomock.Any(), gomock.Any(), gomock.Any()).Return(&linodego.ObjectStorageBucket{
+					Label: "test-bucket",
+				}, nil)
+			},
+			expectedError: nil,
+		},
+		{
+			name: "Error - Unable to get the OBJ bucket",
+			bScope: &scope.ObjectStorageBucketScope{
+				Bucket: &infrav1alpha1.LinodeObjectStorageBucket{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-bucket",
+					},
+					Spec: infrav1alpha1.LinodeObjectStorageBucketSpec{
+						Cluster: "test-cluster",
+					},
+				},
+			},
+			want: nil,
+			expects: func(c *mock.MockLinodeObjectStorageClient) {
+				c.EXPECT().GetObjectStorageBucket(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, fmt.Errorf("error in getting object storage bucket"))
+			},
+			expectedError: fmt.Errorf("failed to get bucket from cluster"),
+		},
+		{
+			name: "Success - Successfully create the OBJ bucket",
+			bScope: &scope.ObjectStorageBucketScope{
+				Bucket: &infrav1alpha1.LinodeObjectStorageBucket{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-bucket",
+					},
+					Spec: infrav1alpha1.LinodeObjectStorageBucketSpec{
+						Cluster: "test-cluster",
+					},
+				},
+			},
+			want: &linodego.ObjectStorageBucket{
+				Label: "test-bucket",
+			},
+			expects: func(c *mock.MockLinodeObjectStorageClient) {
+				c.EXPECT().GetObjectStorageBucket(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, nil)
+				c.EXPECT().CreateObjectStorageBucket(gomock.Any(), gomock.Any()).Return(&linodego.ObjectStorageBucket{
+					Label: "test-bucket",
+				}, nil)
+			},
+			expectedError: nil,
```

Comment:
You can omit this line, as `expectedError` will already be set to `nil` by default.